var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -264,6 +267,108 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
 
     build_directory = 'spack-build'
 
+    @property
+    def executables(self):
+        names = [r'gcc', r'[^\w]?g\+\+', r'gfortran']
+        suffixes = [r'-mp-\d\.\d', r'-\d\.\d', r'-\d', r'\d\d']
+        return [r''.join(x) for x in itertools.product(names, suffixes)]
+
+    @classmethod
+    def filter_detected_exes(cls, prefix, exes_in_prefix):
+        result = []
+        for exe in exes_in_prefix:
+            # clang++ matches g++ -> clan[g++]
+            if 'clang' in exe:
+                continue
```

Comment:
Again, this won't be necessary if we add `^` to all regexes.

Reply:
In this particular case using `^` wouldn't match all the possible compilers, see the configuration file in https://github.com/spack/spack/pull/16526#issuecomment-628919889 as a use case.